pkg/console/controllers/route/controller.go

```diff
@@ -110,59 +122,151 @@ func (c *RouteSyncController) sync() error {
-		return nil, false, "FailedCreate", rtErr
+	defaultRoute, defaultRouteChanged, defaultRouteError := routesub.ApplyRoute(c.routeClient, c.recorder, requiredDefaultRoute)
+	if defaultRouteError != nil {
+		return nil, false, "FailedApply", defaultRouteError
 	}
-	// Check if the console is reachable
-	c.CheckRouteHealth(operatorConfig, rt)
-
-	// we will not proceed until the route is valid. this eliminates complexity with the
-	// configmap, secret & oauth client as they can be certain they have a host if we pass this point.
-	host := routesub.GetCanonicalHost(rt)
-	if len(host) == 0 {
-		return nil, false, "FailedHost", customerrors.NewSyncError(fmt.Sprintf("route is not available at canonical host %s", rt.Status.Ingress))
+
+	return defaultRoute, defaultRouteChanged, "", defaultRouteError
+}
+
+// Custom route sync needs to:
+// 1. validate if the reference for secret with serving-cert is defined in operator config(in case a non-openshift cluster domain is used)
+// 2. if secret is defined, validate if the secret contains TLS certificate and key
+// 3. if secret is defined, get the TLS certificate and key by bese64 decoding them
+// 4. create the custom console route, if custom TLS certificate and key are defined use them
+// 5. validate and stomp the route
+func (c *RouteSyncController) SyncCustomRoute(operatorConfig *operatorsv1.Console) (consoleRoute *routev1.Route, isNew bool, reason string, err error) {
+	if !routesub.IsCustomRouteSet(operatorConfig) {
+		return nil, false, "", nil
+	}
+
+	customSecret, configErr := c.ValidateCustomRouteConfig(operatorConfig)
+	if configErr != nil {
+		return nil, true, "InvalidCustomRouteConfig", configErr
+	}
+
+	customTLSCert, secretValidationErr := c.ValidateCustomCertSecret(customSecret)
+	if secretValidationErr != nil {
+		return nil, true, "InvalidCustomCertificateSecret", secretValidationErr
+	}
+
+	requiredCustomRoute := routesub.CustomRoute(operatorConfig, customTLSCert)
+
+	customRoute, customRouteChanged, customRouteError := routesub.ApplyRoute(c.routeClient, c.recorder, requiredCustomRoute)
+	if customRouteError != nil {
+		return nil, false, "FailedApply", customRouteError
 	}
 
-	if validatedRoute, changed := routesub.Validate(rt); changed {
-		// if validation changed the route, issue an update
-		if _, err := c.routeClient.Routes(api.TargetNamespace).Update(c.ctx, validatedRoute, metav1.UpdateOptions{}); err != nil {
-			// error is unexpected, this is a real error
-			return nil, false, "InvalidRouteCorrection", err
+	return customRoute, customRouteChanged, "", customRouteError
+}
+
+func (c *RouteSyncController) ValidateCustomRouteConfig(operatorConfig *operatorsv1.Console) (*corev1.Secret, error) {
+	// get ingress
+	ingress, err := c.ingressClient.Get(c.ctx, api.ConfigResourceName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	// Check if the custom hostname has cluster domain suffix. If it does,
+	// then the secret that contains TLS certificate and key needs to be
```

Comment:
I'm a little confused by the comment, it implies something should be done, but it immediately returns `nil, nil`.

Reply:
but the user may still want their certificate to be used instead of the ones that the router adds by default, correct?